src/router/upload/index.ts

```diff
@@ -0,0 +1,85 @@
+import { Request, Response, Router } from 'express';
+import Busboy from 'busboy';
+import { join } from 'node:path';
+import { Writable } from 'node:stream';
+import { appendFile } from 'node:fs/promises';
+import { srcPath } from '../../utils/path-helper';
+import { ensureDirectoryExists } from '../../utils/file-helper';
+import { RequestStatusCodes } from '../../utils/request-status-codes';
+
+const uploadRouter = Router();
+
+const FileWritableStream = (filepath: string): Writable => {
+  return new Writable({
+    write: async function (chunk, encoding, next) {
+      try {
+        await appendFile(filepath, chunk);
+        next();
+      } catch (e: unknown) {
+        next(e as Error);
+      }
+    }
+  });
+};
+
+const maxFileSize: number = 500 * 1024 * 1024;
+const validMimeTypes = ['video/mp4'];
+
+uploadRouter.post('/', async (req: Request, res: Response) => {
+  try {
+    if (Number(req.headers['content-length']) > maxFileSize) {
```

Comment:
This is what I said not to do  =)

Reply:
You can do in this way for example

let contentLength = 0; // Initialize content length counter
const maxSizeBytes = 19000000; // --- in bytes ≈ 19 mb (just don't find a bigger file =)) 
    
 req.on('data', (chunk) => {
        // Accumulate the size of each chunk
        contentLength += chunk.length;

        console.log({contentLength})
        if (contentLength > maxSizeBytes) {
            // If the accumulated size exceeds the maximum limit, return an error
            res.status(400).send('File size exceeds the limit');
            req.destroy(); // Close the request stream to stop receiving data
        }
    });